src/main/webapp/meal-page-script.js

```diff
@@ -95,10 +95,39 @@ function createMap() {
       }
     })
     .then(() => {
-      addRestaurants(map);
+      const request = {
+        location: map.getCenter(),
+        radius: "1500",
+        type: ["restaurant", "cafe", "meal_takeaway", "bar", "bakery"],
+        query: type,
+      };
+      const service = new google.maps.places.PlacesService(map);
+      getSearchPromise(service, request, map).then((results) => {
```

Comment:
It kind of depends on a use case, but in general it's a good idea to `return` nested promises.

In this case, the top level promise (the one started by `getCurrentPositionPromise` returns nothing. It'll always resolve with `undefined`. If `getSearchPromise` fails, it'll only log to `console.log`, but you won't be able to do anything with that fact.

However, if you `return getSearchPromise(..)` here, the result will be attached to the parent promise. You'd then be able to `catch()` it in line 108, if need be.

I'd suggest:
1. `return`ing the `getSearchPromise` result
2. `catch`ing the result and logging to `console.log` if something went wrong. It's usually a good idea to leave some trace of errors.